### DISCUSSION TITLE
<Highlight /> with regex

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#2776] [Feature requests]
In our case we need to highlight hashtags like this:
![Screenshot 2022-10-23 at 04 44 37](https://user-images.githubusercontent.com/22228431/197369171-06e0fef0-8b37-49c7-b3ed-e45bf149870c.png)

`Mark` or `Highlight` would do the job but they only accept lists:
![Screenshot 2022-10-23 at 04 46 51](https://user-images.githubusercontent.com/22228431/197369201-0df20a1b-53fd-4dde-95b5-027165efb160.png)

We'd really need regex to achieve this I think.

**Comment by rtivital** [comment]
It is not a good idea to use Highlight for that, usually in such cases you should inject extra markup to HTML string before rendering it

**Reply by atumas-bananamilk** [reply]
The problem with injecting is we post that text to social media channels for users, and text sometimes gets messed up with random HTML.

**Comment by atumas-bananamilk** [comment]
Can we at least get some improvement on Textarea like this?

https://bonafideduck.github.io/react-highlight-within-textarea

**Reply by rtivital** [reply]
You can do that on your side, it should not be a part of the library

**Comment by yukidaruma** [comment]
Hi,

While I understand this request has been dismissed as "not a good idea in such cases", I’d like to point out some use cases where this suggestion could potentially be useful, hoping to encourage further discussion.  

1. **Extending character equivalence to other languages, e.g., "あ" to "ア" in Japanese**
   While matching "A" to "a" is [already supported for Latin text](https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Highlight/highlighter/highlighter.ts#L31), there's no equivalent handling for Japanese Hiragana and Katakana.
   In Japanese search contexts, it’s common to treat "あ" (Hiragana) and "ア" (Katakana) as equivalent, as users often expect script-agnostic results.

   Currently, you can achieve this with `highlight={["あ", "ア"]}`, but this approach becomes tedious when patterns are longer or more complex. (I'm not familiar with other languages beyond Japanese, but a similar need might apply to diacritics like "á" in other scripts)
 
2. **Limiting highlight to the prefix**
   It could be helpful to highlight only the matching prefix portion of a word rather than other occurrences too.
   
   For example, if a user searches for "a," highlighting just the initial "a" in "application" (instead of the second "a" as well) could make the result more intuitive and visually precise.

Additionally, I think supporting a user-defined function would be more versatile than limiting it to regex.
